app/jobs/materialize_permissions.rb

```diff
@@ -0,0 +1,68 @@
+# A job the update a materialized of which users
+# have access to which clients.
+# At this point all it does is walk the existing scopes
+# and output timing data
+
+require 'ruby-progressbar'
+require 'pp'
+require 'benchmark'
+
+class MaterializePermissions < BaseJob
+  COLS = ['user_id', 'client_id', 'viewable'].freeze
+
+  def perform
+    # WIP:
+    # TODO: figure out how to handle other Client scopes...
+    # TODO: what about (soft) deletes of both users and clients
+    # TODO: do we add FK to the UserClientPermission model (which will slow things down a lot)
+    # TODO: do we keep "deleted_at" records to provide audit history?
```

Comment:
This would be ok if we do it in such a way that we aren't completely re-writing the table nightly.

Reply:
Not sure how necessary it is though.  We log access to client records, which is more important.